### PR TITLE
snapdtool, many: support additional key/value flags in info file

### PR DESCRIPTION
**Pull request by anonymouse64**
Also set a new key, SNAPD_APPARMOR_REEXEC, to 0 for now, since this key will be
used shortly with the new apparmor vendoring inside the snapd snap.

cc @alexmurray this is the implementation suggested by Samuele in the other PR, you can
check have mkversion.sh in your branch in https://github.com/snapcore/snapd/pull/11129 write
this flag as 1, and then check for the value "1" in the flags argument to decide if the 
apparmor_parser from the snapd snap should be used or not.

### CODE REVIEW

File: snapdtool/info_file.go [hunk 1]
```diff
@@ -20,34 +20,53 @@
...
-func SnapdVersionFromInfoFile(infoPath string) (string, error) {
-	content, err := ioutil.ReadFile(infoPath)
+// The format of the "info" file are lines with "KEY=VALUE" with the typical key
+// being just VERSION. The file is produced by mkversion.sh and normally
+// installed along snapd binary in /usr/lib/snapd.
+// Other typical keys in this file include SNAPD_APPARMOR_REEXEC, which
+// indicates whether or not the snapd-apparmor binary installed via the
+// traditional linux package of snapd supports re-exec into the version in the
+// snapd or core snaps.
+func SnapdVersionFromInfoFile(infoPath string) (version string, flags map[string]string, err error) {
```
Comment (pedronis) [inline snapdtool/info_file.go:38]: given we are changing this - as I mentioned - it's probably worth change the argument to be the dir, we don't have a use case for the filename itself != "info"

**Review by pedronis — APPROVED** [review]
lgtm with a suggestion

File: snapdtool/info_file.go (hunk 1 above)
Comment (anonymouse64) [reply snapdtool/info_file.go]: done

File: mkversion.sh [hunk 2]
```diff
@@ -130,4 +130,5 @@ EOF
 
 cat <<EOF > "$PKG_BUILDDIR/data/info"
 VERSION=$v
+SNAPD_APPARMOR_REEXEC=0
```
Comment (bboozzoo) [inline mkversion.sh:133]: Hmm I'm almost ready to suggest adding SNAP_MOUNT_DIR at some point to have a reasonable fallback for incorrectly guessed system specific paths.

**Review by bboozzoo — APPROVED** [review]
LGTM

File: snapdtool/info_file.go [hunk 3]
```diff
@@ -35,7 +36,8 @@ import (
 // indicates whether or not the snapd-apparmor binary installed via the
```
Comment (pedronis) [inline snapdtool/info_file.go:36]: the doc comment needs adjusting as it mentions still infoPath.

**Review by pedronis — COMMENTED** [review]
still looks good but doc comment needs a tweak now

File: mkversion.sh (hunk 2 above)
Comment (anonymouse64) [reply mkversion.sh:133]: that's an interesting idea

File: snapdtool/info_file.go (hunk 3 above)
Comment (anonymouse64) [reply snapdtool/info_file.go:36]: done

**Pull request merged**